diff --git a/includes/entityconnect.form.inc b/includes/entityconnect.form.inc
index decc8a8..11d4932 100644
--- a/includes/entityconnect.form.inc
+++ b/includes/entityconnect.form.inc
@@ -80,7 +80,7 @@ function _entityconnect_field_attach_form($entity_type, $entity, &$form, &$form_
             $acceptable_types = $views_display['default']->display_options['filters']['machine_name']['value'];
           }
           break;
-        
+
         default:
           break;
       }
@@ -420,11 +420,16 @@ function entityconnect_return_form_alter(&$form, &$form_state, $form_id, $cache_
       $language = isset($field_container['#language']) ? $field_container['#language'] : LANGUAGE_NONE;
     }
 
-    isset($language) ? $widget_container = &$field_container[$language] : $widget_container = $field_container;
+    // If no language or language element is empty, use the full container.
+    isset($language) && ! empty($field_container[$language]) ? $widget_container = &$field_container[$language] : $widget_container = $field_container;
     isset($widget_container['#type']) ? $widget_container_type = $widget_container['#type'] : $widget_container_type = 'autocomplete';
 
     $field_info = field_info_field($data['field']);
 
+    if ( $widget_container_type == 'autocomplete' && $widget_container['target_id']['#type'] == 'select' ) {
+      $widget_container_type = 'multiple_selects';
+    }
+
     if (isset($data['target_id']) && $form_state['rebuild'] === TRUE) {
       switch ($data['target_entity_type']) {
         case 'node':
@@ -463,6 +468,10 @@ function entityconnect_return_form_alter(&$form, &$form_state, $form_id, $cache_
                 $element['#default_value'] = $widget_container['#value'] + array($element['#default_value'] => $element['#default_value']);
                 break;
 
+              case 'multiple_selects':
+                $element['#default_value'] = $node->nid;
+                break;
+
               default:
                 if ($field_info['module'] == 'entityreference') {
                   $element['#default_value'] = $node->title;
@@ -513,6 +522,11 @@ function entityconnect_return_form_alter(&$form, &$form_state, $form_id, $cache_
                 $element['#default_value'] = $form[$data['field']][$language]['#value'] + array($element['#default_value'] => $element['#default_value']);
                 break;
 
+              // Multiple selects widget
+              case 'multiple_selects':
+                $element['#default_value'] = $user->uid;
+                break;
+
               default:
                 if ($field_info['module'] == 'entityreference') {
                   $element['#default_value'] = $user->name;
@@ -562,6 +576,11 @@ function entityconnect_return_form_alter(&$form, &$form_state, $form_id, $cache_
                 $element['#default_value'] = $form[$data['field']][$language]['#value'] + array($element['#default_value'] => $element['#default_value']);
                 break;
 
+              // Multiple selects widget
+              case 'multiple_selects':
+                $element['#default_value'] = $term->tid;
+                break;
+
               default:
                 if ($field_info['module'] == 'entityreference') {
                   $element['#default_value'] = $term->name;
@@ -606,6 +625,11 @@ function entityconnect_return_form_alter(&$form, &$form_state, $form_id, $cache_
                 $element['#default_value'] = $form[$data['field']][$language]['#value'] + array($element['#default_value'] => $element['#default_value']);
                 break;
 
+              // Multiple selects widget
+              case 'multiple_selects':
+                $element['#default_value'] = $vocabulary->vid;
+                break;
+
               default:
                 if ($field_info['module'] == 'entityreference') {
                   $element['#default_value'] = $vocabulary->name;
@@ -706,6 +730,7 @@ function entityconnect_child_form_alter(&$form, &$form_state, $form_id, $cache_i
 function _entityconnect_alter_form_state_input(&$form_state, $module, $widget_type, $parents, $language, $element) {
   switch ($widget_type) {
     case 'autocomplete':
+    case 'multiple_selects':
       switch ($module) {
         case 'entityreference':
           array_push($parents, "target_id");
